pkgs/build-support/node/build-npm-package/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchNpmDeps, callPackage, nodejs }:
+
+{ name ? "${args.pname}-${args.version}"
+, src ? null
+, srcs ? null
+, sourceRoot ? null
+, patches ? [ ]
+, npmDepsHash ? ""
+, nativeBuildInputs ? [ ]
+, buildInputs ? [ ]
+, ...
+} @ args:
+
+assert npmDepsHash != "";
+
+let
+  npmDeps = fetchNpmDeps {
+    inherit src srcs sourceRoot patches;
+    name = "${name}-npm-deps";
+    hash = npmDepsHash;
+  };
+
+  # TODO: how do we expose these like Rust? `npmHooks`?
+  inherit (callPackage ./hooks { inherit nodejs; }) npmConfigHook npmBuildHook npmInstallHook;
+in
+stdenv.mkDerivation (args // {
+  inherit npmDeps;
+
+  nativeBuildInputs = nativeBuildInputs ++ [ npmConfigHook npmBuildHook npmInstallHook ];
+  buildInputs = buildInputs ++ [ nodejs ];
+
+  strictDeps = true;
+
+  passthru = { inherit npmDeps; } // (args.passthru or { }); # TODO: why args.passthru or instead of aliasing it above like the others?
```

Comment:
I saw this in [`buildRustPackage`](https://github.com/NixOS/nixpkgs/blob/3ac8876ca335877e9d557aec2d8dd1923ee5bbab/pkgs/build-support/rust/build-rust-package/default.nix#L153), and nothing stands out.

To clarify, I mean why this isn't done:
```nix
{ passthru ? { } }: { passthru = { inherit npmDeps; } // (passthru or { }); }
```

`buildRustPackage` doesn't use a recursive set -- maybe it did at some point and this is a leftover from that?